packages/snaps-controllers/src/snaps/endowments/caveats/requestTime.ts

```diff
@@ -0,0 +1,129 @@
+import type {
+  Caveat,
+  CaveatSpecificationConstraint,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { MaxRequestTimeStruct, SnapCaveatType } from '@metamask/snaps-utils';
+import type { AssertionErrorConstructor, Json } from '@metamask/utils';
+import { assertStruct, hasProperty, isPlainObject } from '@metamask/utils';
+
+import type { CaveatMapperFunction, CaveatMapperReturnValue } from './generic';
+
+/**
+ * Asserts that the given value is a valid `maxRequestTime` value.
+ *
+ * @param value - The value to assert.
+ * @param ErrorWrapper - An optional error wrapper to use. Defaults to
+ * {@link AssertionError}.
+ * @throws If the value is not a valid `maxRequestTime` value.
+ */
+function assertIsMaxRequestTime(
+  value: unknown, // eslint-disable-next-line @typescript-eslint/naming-convention
+  ErrorWrapper?: AssertionErrorConstructor,
+): asserts value is number {
+  assertStruct(
+    value,
+    MaxRequestTimeStruct,
+    'Invalid JSON-RPC origins',
```

Comment:
Should this error message be something else?

Reply:
Yeah, I just discovered it while writing a missing test. Will fix!